nodedb.go

```diff
@@ -707,21 +756,30 @@ func (ndb *nodeDB) UpdateBranch(node *Node) []byte {
 	return node.hash
 }
 
-func (ndb *nodeDB) SaveOrphans(version int64, orphans []*Node) {
+func (ndb *nodeDB) SaveOrphans(batch dbm.Batch, version int64, orphans []*Node) {
 	ndb.mtx.Lock()
 	defer ndb.mtx.Unlock()
-	version = version - 1
-
-	orphansObj := ndb.heightOrphansMap[version]
-	if orphansObj != nil {
-		orphansObj.orphans = orphans
-	}
-	for _, node := range orphans {
-		ndb.orphanNodeCache[string(node.hash)] = node
-		ndb.uncacheNode(node.hash)
-		delete(ndb.prePersistNodeCache, string(node.hash))
-		node.leftNode = nil
-		node.rightNode = nil
+
+	if EnableOptPruing {
```

Comment:
if !EnableOptPruning

Reply:
!EnableOptPruning means run the raw save process